src/utils/get.ts

```diff
@@ -0,0 +1,140 @@
+   const type = typeof value;
+
+   return (type === 'symbol') || (
+      type === 'object' &&
+      value !== undefined &&
+      value !== null &&
+      getTagString(value) === '[object Symbol]'
+   );
+}
+
+function toKey(value: unknown): string | symbol {
+   if (typeof value === 'string' || isSymbol(value)) {
+      return value;
+   }
+
+   return Object.is(value, -0) ? '-0' : `${value}`;
+}
+
+const IS_DEEP_PROP_REGEX = /\.|\[(?:[^[\]]*|(["'])(?:(?!\1)[^\\]|\\.)*?\1)\]/,
+      IS_PLAIN_PROP_REGEX = /^\w*$/;
+
+function isKey(value: any, object: any): boolean {
+   if (Array.isArray(value)) {
+      return false;
+   }
+   const type = typeof value;
+
+   if (type === 'number' || type === 'boolean' || value === null || value === undefined || isSymbol(value)) {
+      return true;
+   }
+   return IS_PLAIN_PROP_REGEX.test(value) || !IS_DEEP_PROP_REGEX.test(value) ||
+      (object !== null && object !== undefined && value in Object(object));
+}
+
+const DOT_CHAR_CODE = '.'.charCodeAt(0),
+      ESCAPE_CHAR_REGEX = /\\(\\)?/g;
+
+/* eslint-disable no-useless-concat */
+const PROP_NAME_REGEX = RegExp(
+   // Match anything that isn't a dot or bracket.
+   '[^.[\\]]+' + '|' +
+   // Or match property names within brackets.
+   '\\[(?:' +
+      // Match a non-string expression.
+      '([^"\'][^[]*)' + '|' +
+      // Or match strings (supports escaping characters).
+      '(["\'])((?:(?!\\2)[^\\\\]|\\\\.)*?)\\2' +
+   ')\\]' + '|' +
+   // Or match "" as the space between consecutive dots or empty brackets.
+   '(?=(?:\\.|\\[\\])(?:\\.|\\[\\]|$))',
+   'g'
+);
+/* eslint-enable no-useless-concat */
+
+function stringToPath(str: string): string[] {
+   const result: string[] = [];
+
+   if (str.charCodeAt(0) === DOT_CHAR_CODE) {
+      result.push('');
+   }
+
+   str.replace(PROP_NAME_REGEX, (match, expression, quote, subString) => {
+      let key = match;
+
+      if (quote) {
+         key = subString.replace(ESCAPE_CHAR_REGEX, '$1');
+      } else if (expression) {
+         key = expression.trim();
+      }
+
+      result.push(key);
+
+      // The types for this callback require us to return a string, but we don't really
+      // care about the result of the call to `replace`, so we just return an empty string
+      // here.
+      return '';
+   });
+
+   return result;
+}
+
+function createPathArray(value: any, object: any): string[] {
+   if (Array.isArray(value)) {
+      return value;
+   }
+
+   return isKey(value, object) ? [ value ] : stringToPath(value);
+}
+
+function get<TObject extends object, TKey extends keyof TObject>(
+   obj: TObject,
+   path: TKey | [TKey],
+   defaultValue?: TObject[TKey]
+): TObject[TKey] | undefined;
+function get<TObject extends StringUnknownMap, TResult = unknown>(
+   obj: TObject,
+   path: string | string[],
+   defaultValue?: TResult
+): TResult | undefined;
+function get<TResult = unknown>(obj: any, path: string | string[], defaultValue?: TResult): TResult | undefined {
```

Comment:
I stuck with the name "get" for this function because I already know what it does and am used to the name from Underscore/Lodash. I'm open to other naming ideas though if the reviewer thinks it's necessary, since `get` is pretty vague. `getProperty` is a bit more specific, for example.